.ado/scripts/release.mjs

```diff
@@ -0,0 +1,61 @@
+import { releaseChangelog, releasePublish, releaseVersion } from 'nx/release/index.js';
+
+import { spawnSync } from "node:child_process";
+import * as util from "node:util";
+
+async function main(options) {
+    const { workspaceVersion, projectsVersionData } = await releaseVersion({
+        specifier: options.version,
+        dryRun: options["dry-run"],
+        verbose: options.verbose,
+    });
+
+    const newRNMVersion = projectsVersionData["react-native-macos"].newVersion;
+    spawnSync("node", ["scripts/releases/set-rn-artifacts-version.js", "--build-type", "release", "--to-version", newRNMVersion], { encoding: "utf-8" });
+    spawnSync("git", ["add", 
+      "packages/react-native/Libraries/Core/ReactNativeVersion.js",
+      "packages/react-native/React/Base/RCTVersion.m",
+      "packages/react-native/ReactCommon/cxxreact/ReactNativeVersion.h",
+      "packages/react-native/ReactAndroid/src/main/java/com/facebook/react/modules/systeminfo/ReactNativeVersion.java",
+      "packages/react-native/ReactAndroid/gradle.properties"
+    ], { encoding: "utf-8" });
```

Comment:
I can probably spin this bit out into a prepublish.mjs and keep using the standard flow..